pkg/controller/rpaasinstance/rpaasinstance_controller.go

```diff
@@ -354,6 +438,285 @@ func (r *ReconcileRpaasInstance) listDefaultFlavors(ctx context.Context, instanc
+}
+
+func (r *ReconcileRpaasInstance) reconcileSecretForSessionTickets(ctx context.Context, instance *v1alpha1.RpaasInstance) (*corev1.Secret, error) {
+	enabled := isTLSSessionTicketEnabled(instance)
+
+	newSecret, err := newSecretForTLSSessionTickets(instance)
+	if err != nil {
+		return nil, err
+	}
+
+	var secret corev1.Secret
+	secretName := types.NamespacedName{
+		Name:      newSecret.Name,
+		Namespace: newSecret.Namespace,
+	}
+	err = r.client.Get(ctx, secretName, &secret)
+	if err != nil && k8sErrors.IsNotFound(err) {
+		if !enabled {
+			return nil, nil
+		}
+
+		return newSecret, r.client.Create(ctx, newSecret)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !enabled {
+		return nil, r.client.Delete(ctx, &secret)
+	}
+
+	newData := make(map[string][]byte)
+	for k, v := range newSecret.Data {
+		if vv, found := secret.Data[k]; found {
+			newData[k] = vv
+			continue
+		}
+		newData[k] = v
+	}
+
+	for k, v := range secret.Data {
+		if _, found := newSecret.Data[k]; found {
+			newData[k] = v
+		}
+	}
+
+	if !reflect.DeepEqual(newData, secret.Data) {
+		secret.Data = newData
+		return &secret, r.client.Update(ctx, &secret)
+	}
+
+	return &secret, nil
+}
+
+func (r *ReconcileRpaasInstance) reconcileCronJobForSessionTickets(ctx context.Context, instance *v1alpha1.RpaasInstance, secret *corev1.Secret) error {
+	enabled := isTLSSessionTicketEnabled(instance)
+
+	newCronJob := newCronJobForSessionTickets(instance, secret)
+
+	var cj batchv1beta1.CronJob
+	cjName := types.NamespacedName{
+		Name:      newCronJob.Name,
+		Namespace: newCronJob.Namespace,
+	}
+	err := r.client.Get(ctx, cjName, &cj)
+	if err != nil && k8sErrors.IsNotFound(err) {
+		if !enabled {
+			return nil
+		}
+
+		return r.client.Create(ctx, newCronJob)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if !enabled {
+		return r.client.Delete(ctx, &cj)
+	}
+
+	if reflect.DeepEqual(newCronJob.Spec, cj.Spec) {
+		return nil
+	}
+
+	newCronJob.ResourceVersion = cj.ResourceVersion
+	return r.client.Update(ctx, newCronJob)
+}
+
+func newCronJobForSessionTickets(instance *v1alpha1.RpaasInstance, secret *corev1.Secret) *batchv1beta1.CronJob {
+	enabled := isTLSSessionTicketEnabled(instance)
+
+	keyLength := v1alpha1.DefaultSessionTicketKeyLength
+	if enabled && instance.Spec.TLSSessionResumption.SessionTicket.KeyLength != 0 {
+		keyLength = instance.Spec.TLSSessionResumption.SessionTicket.KeyLength
+	}
+
+	rotationInterval := v1alpha1.DefaultSessionTicketKeyRotationInteval
+	if enabled && instance.Spec.TLSSessionResumption.SessionTicket.KeyRotationInterval != 0 {
+		rotationInterval = instance.Spec.TLSSessionResumption.SessionTicket.KeyRotationInterval
+	}
+
+	image := defaultCacheSnapshotCronImage
+	if enabled && instance.Spec.TLSSessionResumption.SessionTicket.Image != "" {
+		image = instance.Spec.TLSSessionResumption.SessionTicket.Image
+	}
+
+	var secretName string
+	if secret != nil {
+		secretName = secret.Name
+	}
+
+	var secretNamespace string
+	if secret != nil {
```

Comment:
I suppose our never create resource in other odd namespace, why this namespace logic here ?, we could ever assume instance.Namespace ?

Reply:
Absolutely.